File: engine/models/tactical-setup.ts
import { TacticalStyle } from "@/engine/enums";
import type { Formation } from "./formation";

/** Full tactical setup for a match: formation, style, and lineup. */
export interface TacticalSetup {
  formation: Formation;
  style: TacticalStyle;

  /** Player IDs for the starting 11, matching formation.positions order. */
  startingPlayerIds: readonly number[];
}

File: src/data/Models/TacticalSetup.cs
using ElevenLegends.Data.Enums;

namespace ElevenLegends.Data.Models;

/// <summary>
/// Full tactical setup for a match: formation, style, and lineup.
/// </summary>
public sealed record TacticalSetup
{
    public required Formation Formation { get; init; }
    public TacticalStyle Style { get; init; } = TacticalStyle.Balanced;

    /// <summary>Player IDs for the starting 11, matching Formation.Positions order.</summary>
    public required IReadOnlyList<int> StartingPlayerIds { get; init; }
}
